tests/test__collection_api.py

```diff
@@ -1406,3 +1406,87 @@ def test_aggregate_unwind_push_first(self):
                 '_id': 'ooo'
             }]
         self.assertEqual(expect, list(actual))
+
+    def test_aggregate_unwind_pushall(self):
+        collection = self.db.collection
+        collection.insert_many(
+            [
+                {
+                    '_id': 1111,
+                    'a': [
+                        {
+                            'class': '03',
+                            'a': [
+                                {
+                                    'b': '030502',
+                                    'weight': 100.0
+                                },
+                                {
+                                    'b': '030207',
+                                    'weight': 100.0
+                                }
+                            ]
+                        }
+                    ],
+                    'id': 'ooo',
+                    'update_time': 1111
+                },
+                {
+                    '_id': 22222,
+                    'a': [
+                        {
+                            'class': '03',
+                            'a': [
+                                {
+                                    'b': '030502',
+                                    'weight': 99.0
+                                },
+                                {
+                                    'b': '0302071',
+                                    'weight': 100.0
+                                }
+                            ]
+                        }
+                    ],
+                    'id': 'ooo',
+                    'update_time': 1222
+                }
+            ]
+        )
+        actual = collection.aggregate(
+            [
+                {'$sort': {'update_time': -1}},
```

Comment:
This is unrelated to the feature you're trying to test. I would prefer to have unit tests specific to the feature that they are testing: if the test fails it's easier to understand what feature is causing it.

If you really want to test this whole features together, please create a separate test for it that you name accordingly.